src/lib.rs

```diff
@@ -186,50 +193,170 @@ decl_module! {
 
 impl<T: Trait> Module<T> {
 
-    pub fn create_class(name: Vec<u8>, description: Vec<u8>, properties: Vec<Property>, schemas: Vec<ClassSchema>) -> Result<ClassId, &'static str> {
-        // TODO impl
+    pub fn create_class(
+        properties: Vec<Property>,
+        schemas: Vec<ClassSchema>,
+        name: Vec<u8>,
+        description: Vec<u8>
+    ) -> Result<ClassId, &'static str> {
+        
+        ensure!(schemas.len() > 0, ERROR_CLASS_EMPTY_SCHEMAS);
+        ensure!(properties.len() > 0, ERROR_CLASS_EMPTY_PROPS);
+        ensure!(name.len() > 0, ERROR_CLASS_EMPTY_NAME);
+        ensure!(description.len() > 0, ERROR_CLASS_EMPTY_DESCRIPTION);
+
+        let class_id = <NextClassId<T>>::get();
+
+        let new_class = Class {
+            id: class_id,
+            properties,
+            schemas,
+            name,
+            description,
+        };
+
+        // Save newly created class:
+        <ClassById<T>>::insert(class_id, new_class);
+
+        // Increment the next class id:
+        <NextClassId<T>>::put(class_id + 1);
+
+        Self::deposit_event(RawEvent::ClassCreated(class_id));
+        Ok(class_id)
+    }
 
-        // let class_id = <NextClassId<T>>::get();
+    pub fn add_class_property(class_id: ClassId, property: Property) -> Result<u16, &'static str> {
 
-        // // Create new category
-        // let new_class = Class {
-        //     // TODO finish
-        // };
+        ensure!(<ClassById<T>>::exists(class_id), ERROR_CLASS_NOT_FOUND);
 
-        // // Insert category in map
-        // <ClassById<T>>::insert(new_class.id, new_class);
+        let mut class = <ClassById<T>>::get(class_id);
+        let prop_idx = class.properties.len() as u16;
+        class.properties.push(property);
+        <ClassById<T>>::insert(class_id, class);
 
-        // // Update other things
-        // <NextClassId<T>>::put(class_id + 1);
+        Self::deposit_event(RawEvent::ClassPropertyAdded(class_id, prop_idx));
+        Ok(prop_idx)
+    }
 
-        // // Generate event
-        // Self::deposit_event(RawEvent::ClassCreated(account.clone(), class_id));
+    pub fn add_class_schema(class_id: ClassId, schema: ClassSchema) -> Result<u16, &'static str> {
+        
+        ensure!(<ClassById<T>>::exists(class_id), ERROR_CLASS_NOT_FOUND);
 
-        Err(ERROR_NOT_IMPLEMENTED)
-    }
+        ensure!(schema.properties.len() > 0, ERROR_NO_PROPS_IN_CLASS_SCHEMA);
+        
+        let mut class = <ClassById<T>>::get(class_id);
 
-    pub fn add_class_property(class_id: ClassId, property: Property) -> Result<u16, &'static str> {
-        // TODO impl
-        Err(ERROR_NOT_IMPLEMENTED)
-    }
+        if let Some(prev_schema) = class.schemas.last() {
```

Comment:
Hm.. ok, I can remove this check. But a rationale behind this check was:
"It doesn't make sense to create a new schema version if it has the same set of properties as the previous schema on a class."

But it could make sense to have several schemas that have the same set of properties in this scenario: 
1. Imagine you have a schema for a video and the first schema has only a `description` property. This is a schema v0.
2. Then you decide to add a `license` property. You create another schema that contains both `description` and `license`. This is a schema v1.
3. Then for some reason, you decide to remove a `license` field. Then you have two options:
  3.1 Create another schema v2 with only `description` field. This new schema has the same set of properties as schema v0.
  3.2 Delete the schema v1 where we have `license` field. But this operation is quite tricky: if you going to delete a schema from the class, then you need to iterate over all entities of this schema and everywhere delete corresponding schema index and properties, that had been used in this schema and those properties could be deleted, that are not used in other schemas of this class.

Reply:
Removed this check. Now it's possible to use multiple schemas with an equal set of properties in the same class.